context/context.go

```diff
@@ -43,6 +46,21 @@ func initOpVersion() {
 	OpVersion = MaxOpVersion
 }
 
+func initEtcdClient() error {
+	Hostname, err := os.Hostname()
```

Comment:
Hostname would be blank when etcd start is triggered as we initialize context post (re)starting etcd